src/System.Private.CoreLib/shared/System/Reflection/SignatureTypeExtensions.cs

```diff
@@ -108,13 +107,14 @@ internal static bool MatchesExactly(this SignatureType pattern, Type actual)
         /// the method we're looking for, we return null rather than let the TypeLoadException bubble up. The DefaultBinder will catch
         /// the null and continue its search for a better candidate.
         /// </summary>
-        internal static Type TryResolveAgainstGenericMethod(this SignatureType signatureType, MethodInfo genericMethod)
+        internal static Type? TryResolveAgainstGenericMethod(this SignatureType signatureType, MethodInfo genericMethod)
         {
             return signatureType.TryResolve(genericMethod.GetGenericArguments());
         }
     
-        private static Type TryResolve(this SignatureType signatureType, Type[] genericMethodParameters)
+        private static Type? TryResolve(this SignatureType signatureType, Type[] genericMethodParameters)
         {
+            Debug.Assert(signatureType.ElementType != null);
```

Comment:
Is this assert always valid?  It looks like ElementType is only dereferenced one some of the if/else if branches below.

Reply:
There is 4 rows accessing signatureType.ElementType without null check and producing warning, thought its better use assert instead of adding 4 '!'. But yeah agree with your point, will be removed